Marlin/src/pins/stm32f4/pins_BTT_SKR_PRO_common.h

```diff
@@ -237,21 +237,77 @@
 #define TEMP_2_PIN                          PF6   // T3 <-> E2
 #define TEMP_BED_PIN                        PF3   // T0 <-> Bed
 
+// define TEMP_CHAMBER_PIN and TEMP_PROBE_PIN if needed and possible 
+#if HOTENDS == 2
+  #if TEMP_SENSOR_PROBE
+    #define TEMP_PROBE_PIN            TEMP_2_PIN
+  #elif TEMP_SENSOR_CHAMBER
+    #define TEMP_CHAMBER_PIN          TEMP_2_PIN  // use T3 for Chamber sensor
+  #endif
+#endif
+#if HOTENDS < 2
+  #if TEMP_SENSOR_PROBE
+    #define TEMP_PROBE_PIN            TEMP_1_PIN  // use T2 for Probe
+  #endif
+  #if TEMP_SENSOR_CHAMBER
+    #define TEMP_CHAMBER_PIN          TEMP_2_PIN  // use T3 for Chamber sensor
+  #endif
+#endif
+
+// change Pins to Pins without pullup, if a Sensor that doesnt need pullup is selected
+// select ADC pins without pullup, if Sensor Type needs input without pullup
+#if TEMP_SENSOR_0 == -4 || TEMP_SENSOR_0 == 20
+  #define TEMP_0_PIN PF8
+#endif
+#if TEMP_SENSOR_1 == -4 || TEMP_SENSOR_1 == 20
+  #define TEMP_1_PIN PF9
+#endif
+#if TEMP_SENSOR_2 == -4 || TEMP_SENSOR_2 == 20
+  #define TEMP_0_PIN PF10
+#endif
+#if TEMP_SENSOR_BED == -4 || TEMP_SENSOR_BED == 20
+  #define TEMP_BED_PIN PF7
+#endif
+#if (TEMP_SENSOR_PROBE == -4 || TEMP_SENSOR_PROBE == 20) && TEMP_PROBE_PIN
+  #if HOTENDS == 2
+    #define TEMP_PROBE_PIN PF10
```

Comment:
#if … TEMP_PROBE_PIN … define TEMP_PROBE_PIN ?

Reply:
In principle yes, but am not sure this is from me..
The idea of the section is to select a non pull-up ADC pin, if there is a sensor type selected, that doesn't need / is incompatible with pull-up.
For Probe and chamber it only makes sense if there was a Probe pin selected at all. 
If no pin was selected automatically before, it doesn't make sense to reassign it. In that case the one wanting to use a Probe has to manually find another suitable pin. Then he may enter it after here.

Reply:
The point of my comment is to point out that an `#undef` is needed, or there should be an `#else` block with the default define moved into it.

Reply:
I rechecked code and yes with a undef it is nicer.. will add it, if I can